src/BotPlutusInterface/Contract.hs

```diff
@@ -180,38 +200,92 @@ handlePABReq contractEnv req = do
-    WriteBalancedTxReq tx ->
-      WriteBalancedTxResp <$> writeBalancedTx @w contractEnv tx
+    WriteBalancedTxReq tx' ->
+      WriteBalancedTxResp <$> writeBalancedTx @w contractEnv tx'
     AwaitSlotReq s -> AwaitSlotResp <$> awaitSlot @w contractEnv s
     AwaitTimeReq t -> AwaitTimeResp <$> awaitTime @w contractEnv t
-    CurrentSlotReq -> CurrentSlotResp <$> currentSlot @w contractEnv
+    CurrentPABSlotReq -> CurrentPABSlotResp <$> currentSlot @w contractEnv
     CurrentTimeReq -> CurrentTimeResp <$> currentTime @w contractEnv
     PosixTimeRangeToContainedSlotRangeReq posixTimeRange ->
       either (error . show) (PosixTimeRangeToContainedSlotRangeResp . Right)
         <$> posixTimeRangeToContainedSlotRange @w posixTimeRange
     AwaitTxStatusChangeReq txId -> AwaitTxStatusChangeResp txId <$> awaitTxStatusChange @w contractEnv txId
+    AdjustUnbalancedTxReq unbalancedTx -> AdjustUnbalancedTxResp <$> adjustUnbalancedTx' @w contractEnv unbalancedTx
     ------------------------
     -- Unhandled requests --
     ------------------------
-    -- AwaitTimeReq t -> pure $ AwaitTimeResp t
-    -- AwaitUtxoSpentReq txOutRef -> pure $ AwaitUtxoSpentResp ChainIndexTx
-    -- AwaitUtxoProducedReq Address -> pure $ AwaitUtxoProducedResp (NonEmpty ChainIndexTx)
-    -- AwaitTxOutStatusChangeReq TxOutRef
-    -- ExposeEndpointReq ActiveEndpoint -> ExposeEndpointResp EndpointDescription (EndpointValue JSON.Value)
-    -- YieldUnbalancedTxReq UnbalancedTx
-    unsupported -> error ("Unsupported PAB effect: " ++ show unsupported)
+    AwaitUtxoSpentReq _ -> error ("Unsupported PAB effect: " ++ show req)
+    AwaitUtxoProducedReq _ -> error ("Unsupported PAB effect: " ++ show req)
+    AwaitTxOutStatusChangeReq _ -> error ("Unsupported PAB effect: " ++ show req)
+    ExposeEndpointReq _ -> error ("Unsupported PAB effect: " ++ show req)
+    YieldUnbalancedTxReq _ -> error ("Unsupported PAB effect: " ++ show req)
+    CurrentChainIndexSlotReq -> error ("Unsupported PAB effect: " ++ show req)
 
   printBpiLog @w (Debug [PABLog]) $ pretty resp
   pure resp
 
+-- do-not-remove yet, could be handy for debugging of "own" implementation below
+-- "own" implementation will be tested with https://github.com/mlabs-haskell/plutip/issues/119
+-- on local network
+-- adjustUnbalancedTx' ::
+--   forall (w :: Type) (effs :: [Type -> Type]).
+--   ContractEnvironment w ->
+--   UnbalancedTx ->
+--   Eff effs (Either Tx.ToCardanoError UnbalancedTx)
+-- adjustUnbalancedTx' contractEnv unbalancedTx = do
+--   let slotConfig = SlotConfig 200 1654524000
+--       networkId = contractEnv.cePABConfig.pcNetwork
+--       maybeParams = contractEnv.cePABConfig.pcProtocolParams >>= \pparams -> pure $ Params slotConfig pparams networkId
+--   case maybeParams of
+--     Just params -> pure $ snd <$> adjustUnbalancedTx params unbalancedTx
+--     _ -> pure . Left $ Tx.TxBodyError "no protocol params"
+
+adjustUnbalancedTx' ::
```

Comment:
For Me: use `MinUtxo` Effect from `NodeQuery` effect to avoid code duplication.

Reply:
Fixed by https://github.com/mlabs-haskell/bot-plutus-interface/pull/103/commits/038c333eba11c0c9a9dd636691850f00c534f661. Though the tests fail, because we don't handle `MinUtxo` case in the `MockContract`, which is [here](https://github.com/mlabs-haskell/bot-plutus-interface/pull/103/files#diff-6ceb0bce2a099e300e50c52a263dbbb51f4477f208e68f3b9900e98207a6cd6eR382)

Reply:
I will fix it later, as this requires a proper way to test `NodeQuery` Effect.